src/sentry/tasks/unmerge.py

```diff
@@ -0,0 +1,544 @@
+        # exclusive access to it (which is not a safe assumption.)
+        GroupHash.objects.filter(
+            project_id=project.id,
+            hash__in=fingerprints,
+        ).update(group=destination_id)
+
+        # Create activity records for the source and destination group.
+        Activity.objects.create(
+            project_id=project.id,
+            group_id=destination_id,
+            type=Activity.UNMERGE_DESTINATION,
+            user_id=actor_id,
+            data={
+                'fingerprints': fingerprints,
+                'source_id': source_id,
+            },
+        )
+
+        Activity.objects.create(
+            project_id=project.id,
+            group_id=source_id,
+            type=Activity.UNMERGE_SOURCE,
+            user_id=actor_id,
+            data={
+                'fingerprints': fingerprints,
+                'destination_id': destination_id,
+            },
+        )
+    else:
+        # Update the existing destination group.
+        destination = Group.objects.get(id=destination_id)
+        destination.update(**get_group_backfill_attributes(caches, destination, events))
+
+    event_id_set = set(event.id for event in events)
+
+    Event.objects.filter(
+        project_id=project.id,
+        id__in=event_id_set,
+    ).update(group_id=destination_id)
+
+    for event in events:
+        event.group = destination
+
+    EventTag.objects.filter(
+        project_id=project.id,
+        event_id__in=event_id_set,
+    ).update(group_id=destination_id)
+
+    event_event_id_set = set(event.event_id for event in events)
+
+    EventMapping.objects.filter(
+        project_id=project.id,
+        event_id__in=event_event_id_set,
+    ).update(group_id=destination_id)
+
+    UserReport.objects.filter(
+        project_id=project.id,
+        event_id__in=event_event_id_set,
+    ).update(group=destination_id)
+
+    return destination.id
+
+
+def truncate_denormalizations(group_id):
+    GroupTagKey.objects.filter(
+        group_id=group_id,
+    ).delete()
+
+    GroupTagValue.objects.filter(
+        group_id=group_id,
+    ).delete()
+
+    GroupRelease.objects.filter(
+        group_id=group_id,
+    ).delete()
+
+    tsdb.delete([
+        tsdb.models.group,
+    ], [group_id])
+
+    tsdb.delete_distinct_counts([
+        tsdb.models.users_affected_by_group,
+    ], [group_id])
+
+    tsdb.delete_frequencies([
+        tsdb.models.frequent_releases_by_group,
+        tsdb.models.frequent_environments_by_group,
+    ], [group_id])
+
+
+def collect_tag_data(events):
+    results = {}
+
+    for event in events:
+        tags = results.setdefault(event.group_id, {})
+
+        for key, value in event.get_tags():
+            values = tags.setdefault(key, {})
+
+            if value in values:
+                times_seen, first_seen, last_seen = values[value]
+                values[value] = (times_seen + 1, event.datetime, last_seen)
```

Comment:
It's not strong at all (in fact it's likely wrong), but this was intentionally written this way to emulate the existing behavior. (There are more details over at GH-5289 if you're interested.)